ext/storage_sources/azure_store/test/test_azure_connection.cpp

```diff
@@ -30,5 +30,176 @@
+    const std::string non_exist_object_key = "test_non_exist";
+    blob_objects.push_back(std::make_pair(object_name, payload));
+    blob_objects.push_back(std::make_pair(object_name_2, payload_2));
+
+    for (auto pair : blob_objects) {
+        auto blob_client = azure_client.GetBlockBlobClient(obj_prefix + pair.first);
+        blob_client.UploadFrom(create_file(pair.first, pair.second));
+    }
+
+    SECTION("Check Azure connection constructor.", "[azure-connection]")
+    {
+        REQUIRE_THROWS_WITH(azure_connection("Bad_bucket", ""), "Bad_bucket : No such bucket.");
+    }
+
+    SECTION("Check object exists in Azure.", "[azure-connection]")
+    {
+        // Object exists so there should be 1 object.
+        REQUIRE(conn.object_exists(object_name, exists) == 0);
+        REQUIRE(exists == true);
+
+        // Object does not exist so there should be 0 objects.
+        REQUIRE(conn.object_exists(non_exist_object_key, exists) == 0);
+        REQUIRE(exists == false);
+    }
+
+    SECTION("List Azure objects under the test bucket.", "[azure-connection]")
+    {
+        // No matching objects. Object size should be 0.
+        objects.clear();
+        REQUIRE(conn.list_objects(obj_prefix + non_exist_object_key, objects, false) == 0);
+        REQUIRE(objects.size() == 0);
+
+        // No matching objects with listSingle. Object size should be 0.
+        REQUIRE(conn.list_objects(obj_prefix + non_exist_object_key, objects, true) == 0);
+        REQUIRE(objects.size() == 0);
+
+        // List all objects. This is the size of blob_objects.
+        REQUIRE(conn.list_objects(obj_prefix + object_name, objects, false) == 0);
+        REQUIRE(objects.size() == blob_objects.size());
+
+        // List single. Object size should be 1.
+        objects.clear();
+        REQUIRE(conn.list_objects(obj_prefix + object_name, objects, true) == 0);
+        REQUIRE(objects.size() == 1);
+    }
+
+    SECTION("Test delete functionality for Azure.", "[azure-connection]")
+    {
+        auto blob_client = azure_client.GetBlockBlobClient(obj_prefix + object_name + "1");
+
+        blob_client.UploadFrom(create_file(object_name + "1", payload));
+
+        // Test that an object can be deleted.
+        REQUIRE(conn.delete_object(object_name + "1") == 0);
+
+        // Test that removing an object that doesn't exist returns a ENOENT.
+        REQUIRE(conn.delete_object(non_exist_object_key) == ENOENT);
+    }
+
+    SECTION("Check put functionality in Azure.", "[azure-connection]")
+    {
+        const std::string path = "./" + create_file(object_name, payload);
+
+        REQUIRE(conn.put_object(object_name + "1", path) == 0);
+
+        auto blob_client = azure_client.GetBlockBlobClient(obj_prefix + object_name + "1");
+
+        // Test that putting an object that doesn't exist locally returns -1.
+        REQUIRE(conn.put_object(non_exist_object_key, non_exist_object_key + ".txt") == -1);
```

Comment:
Can we make sure through the blobClient that the object doesn't exist, to make sure that it truly doesn't exist even when we return -1?

Reply:
yup i've added a check to check that the object does not exist when put fails

Reply:
https://github.com/wiredtiger/wiredtiger/pull/8747/commits/27d5d3c123e9a19501ffdc1d27fccbadc121470e